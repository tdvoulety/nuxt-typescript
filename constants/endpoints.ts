export const COMPANIES_URL: string = '/companies'
export const SEARCH_APP_URL: string = '/search_app/api/v1/'
export const BOOKMARKS_URL: string = '/bookmarks/api/v1/company_bookmarks'
export const TRACKING_EVENTS_URL: string = '/octopus/api/events'
export const TRACKING_PAGES_URL: string = '/octopus/api/pages'
export const SWITCHER_URL: string = '/internal_api/v1/flags?only='
