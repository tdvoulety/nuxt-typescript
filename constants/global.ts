interface Component {
  key: string,
  component: object
}

export const GLOBAL_COMPONENTS: Component[] = []

export const GLOBAL_CLICK_EVENT: string = 'GLOBAL_CLICK_EVENT'
export const GLOBAL_RESIZE_EVENT: string = 'GLOBAL_RESIZE_EVENT'
