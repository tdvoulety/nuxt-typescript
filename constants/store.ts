export const NAMESPACES = {
  I18N: 'i18n/',
  AUTH: 'auth/',
}

export const ACTIONS = {
  NUXT_SERVER_INIT: 'nuxtServerInit',
  INIT: 'INIT',
  GET_ITEMS: 'GET_ITEMS',
  GET_DETAIL: 'GET_DETAIL',
}

export const MUTATIONS = {
  SET_BASE: 'SET_BASE',
  SET_LOCALE: 'SET_LOCALE',
  SET_SIDE_NAV_VISIBILITY: 'SET_SIDE_NAV_VISIBILITY',
}
