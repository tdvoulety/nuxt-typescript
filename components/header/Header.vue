<template>
  <header class="app-header">
    <b-container>
      <NavBar
        :sideNavVisible="sideNavVisible"
        @toggleSideNav="toggleSideNav"
      />

      <button type="button" class="app-header__navbar-toggler" @click="toggleSideNav">
        <img :data-src="require('~/assets/img/header/mobile-toggle.svg')" alt="" title="" class="lazyload"/>
      </button>

      <a :href="$t('routes./home')" class="app-header__logo">
        <img :data-src="require('~/assets/img/header/logo.svg')" alt="" title="" class="lazyload" />
      </a>
    </b-container>
  </header>
</template>

<script lang="ts">
import { MUTATIONS } from '@/constants/store'

export default {
  components: {
    NavBar: () => import('@/components/header/NavBar.vue'),
  },
  computed: {
    sideNavVisible(): any {
      // @ts-ignore
      return this.$store.state.sideNavVisible
    },
  },
  methods: {
    toggleSideNav(): void {
      // @ts-ignore
      this.$store.commit(MUTATIONS.SET_SIDE_NAV_VISIBILITY, !this.sideNavVisible)
    },
  },
}
</script>

<style lang="scss" scoped>
.app-header {
  display: block;
  background: #aaffc4;

  &__navbar-toggler {
    position: absolute;
    right: .8rem;
    top: -.3rem;
    width: 6.2rem;
    height: 6rem;
    line-height: 6rem;
    padding: 0;
    background-color: transparent;
    border: 0;
    color: grey;
    cursor: pointer;

    &:hover, &:focus {
      outline: none !important; // stylelint-disable-line declaration-no-important
      box-shadow: none !important; // stylelint-disable-line declaration-no-important
    }

    @include media-breakpoint-up(md) {
      display: none;
      top: 0;
      right: 0;
    }
  }

  &__logo {
    font-size: 4rem;
    height: 1em;
    width: 1.5em;
    padding: 0;

    @include media-breakpoint-up(md) {
      font-size: 6.667rem;
      order: 1;
      top: 5rem;
    }

    img {
      display: block;
      height: inherit;
      width: inherit;
    }
  }
}
</style>
