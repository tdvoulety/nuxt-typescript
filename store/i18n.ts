import { MUTATIONS } from '@/constants/store'

export const state = () => ({
  locale: null,
})

export const mutations = {
  // @ts-ignore
  [MUTATIONS.SET_LOCALE](state, payload = null) {
    state.locale = payload
  },
}
