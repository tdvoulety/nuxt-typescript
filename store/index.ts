import config from '@/config/env'
import { ACTIONS, MUTATIONS } from '@/constants/store'

export const state = () => ({
  config,
  base: null,
  sideNavVisible: false,
})

export const actions = {
  // @ts-ignore
  async [ACTIONS.NUXT_SERVER_INIT]({ commit }, { req }) {
    const headers = req && req.headers
    const domain = headers && (headers['x-forwarded-host'] || headers.host)

    commit(MUTATIONS.SET_BASE, `https://${domain}`)

    return Promise.resolve()
  },
}

export const mutations = {
  // @ts-ignore
  [MUTATIONS.SET_BASE](state, payload = null) {
    state.base = payload
  },
  // @ts-ignore
  [MUTATIONS.SET_SIDE_NAV_VISIBILITY](state, payload = false) {
    state.sideNavVisible = payload
  },
}
