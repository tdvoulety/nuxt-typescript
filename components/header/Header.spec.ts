// @ts-nocheck
import Header from '@/components/header/Header'
import { MUTATIONS } from '@/constants/store'
import { mountComponent, rendersToSnapshot } from '~/test/unit/helpers'

const mount = () => mountComponent(Header, {
  state: {
    sideNavVisible: false,
    auth: {
      user: null,
    },
  },
})

describe('Header', () => {
  it('render', () => {
    const { wrapper } = mount()

    expect(wrapper.findAll('.app-header').length).toEqual(1)
    expect(wrapper.findAll('.app-header__navbar-toggler').length).toEqual(1)
    expect(wrapper.findAll('.app-header__logo').length).toEqual(1)
    expect(wrapper.find('.app-header').classes()).not.toContain('app-header--server-error')
  })

  it('commit side navbar mutation after click on toggle button', () => {
    const { wrapper, store } = mount()
    const toggle = wrapper.find('.app-header__navbar-toggler').element

    toggle.click()
    expect(store.commit).toHaveBeenCalledTimes(1)
    expect(store.commit).toHaveBeenCalledWith(MUTATIONS.SET_SIDE_NAV_VISIBILITY, true)
  })

  it('renders to a snapshot', () => {
    const { wrapper } = mount()

    rendersToSnapshot(wrapper)
  })
})
