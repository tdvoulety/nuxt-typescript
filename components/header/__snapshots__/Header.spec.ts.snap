// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders to a snapshot 1`] = `
<header class="app-header">
  <b-container-stub tag="div">
    <NavBar-stub></NavBar-stub> <button type="button" class="app-header__navbar-toggler"><img data-src="" alt="" title="" class="lazyload"></button> <a href="#routes./home#" class="app-header__logo"><img data-src="" alt="" title="" class="lazyload"></a>
  </b-container-stub>
</header>
`;
